src/redux/reducers.tsx

```diff
@@ -44,27 +46,34 @@ export default function(state: AppState = loadState(), action: TActions) {
 function toggleScale(state: AppState): AppState {
   return persistedState({
     ...state,
-    scale: state.scale === EScale.Linear ? EScale.Log : EScale.Linear,
+    scale: state.scale === EScale.Linear ? EScale.Log : EScale.Linear
   });
 }
 
 function changeZoom(state: AppState, data: number): AppState {
   let zoom = state.zoom + data;
-  if (zoom > 5) { zoom = 5; }
-  else if (zoom < 1) { zoom = 1; }
+  if (zoom > 5) {
+    zoom = 5;
+  } else if (zoom < 1) {
+    zoom = 1;
+  }
   return persistedState({ ...state, zoom });
 }
 
 function changeMode(state: AppState, to: string): AppState {
   return persistedState({ ...state, mode: to as mode.EMode });
 }
 
+function setRegions(state: AppState, regions: IRegion[]): AppState {
```

Comment:
I feel like we don't need setRegions (that changes everything at once) - or maybe it's simpler this way? I was thinking addRegion, removeRegion, etc.

Either way, I think for the chart component you don't need to worry about setting the region - the controls should likely be in the x-axis and chart just passively follow the change?

Wdyt?

Reply:
Yea, I have some controls in the chart right now. I think those will be replaced by the x-axis that you are making. 

`setRegions` works right now. But feel free to update w/ `addRegion`, `removeRegion` etc as you develop the x-axis